File: Kintino.CipherConf.Crypto/Services/Algos/KeyGenerator.cs
using Kintino.CipherConf.Crypto.Services.Algos;
namespace Kintino.CipherConf.Crypto.Services.Algos;

internal static class KeyGenerator
{
    public const int KeySize = 32; // 256 bits

    public static byte[] GenerateKey()
    {
        return RNG.GetBytes(KeySize);
    }
}

File: EncryptConfig.Js/src/modules/encryption/src/services/KeyGenerator.ts
import type { IKeyGenerator } from "@abstractions";
import { Rng } from "./Rng";

const keySizeInBytes = 32; // 256 bits

export class KeyGenerator implements IKeyGenerator {
	generateKey() {
		return Rng.createRandomBytes(keySizeInBytes);
	}
}
